zrml/swaps/rpc/src/lib.rs

```diff
@@ -205,4 +215,56 @@ where
             })
             .collect()
     }
+
+    async fn get_asset_spot_prices_for_pool(
+        &self,
+        pool_id: PoolId,
+        with_fees: bool,
+        blocks: Vec<NumberFor<Block>>,
+    ) -> RpcResult<BTreeMap<Asset<MarketId>, Vec<SerdeWrapper<Balance>>>> {
+        let api = self.client.runtime_api();
+        let at = BlockId::hash(self.client.info().best_hash);
+        let pool = api
+            .pool_by_id(&at, pool_id.clone())
+            .map_err(|e| {
+                CallError::Custom(ErrorObject::owned(
+                    Error::RuntimeError.into(),
+                    "Unable to get pool, Runtime trapped.",
+                    Some(e.to_string()),
+                ))
+            })?
+            .map_err(|e| {
+                CallError::Custom(ErrorObject::owned(
+                    Error::RuntimeError.into(),
+                    "Unable to get pool. DispatchError",
+                    Some(format!("{:?}", e)),
+                ))
+            })?;
+        let mut result_map = BTreeMap::<Asset<MarketId>, Vec<SerdeWrapper<Balance>>>::new();
+        for asset in &pool.assets {
+            if asset != &pool.base_asset {
+                let prices: Vec<Result<SerdeWrapper<Balance>, CallError>> = blocks
+                    .clone()
+                    .into_iter()
+                    .map(|block| {
+                        let hash = BlockId::number(block);
+                        let res = api
+                            .get_spot_price(&hash, &pool_id, &asset, &pool.base_asset, with_fees)
```

Comment:
so IIUC using asset as in_asset gives value of input asset given one unit of output asset right?
if we change as you propose it will be how much base asset needed for output asset.

Reply:
> if we change as you propose it will be how much base asset needed for one unit of output asset.

That's correct.